File: BlogApp-RN/src/features/authentication/domain/usecases/logout.ts
import { AsyncResult } from "../../../../core/typedefs/asyncResult";
import { ResponseMessage } from "../../data/models/responseMessage";
import AuthRepo from "../repos/authRepo";

class Logout {
  constructor(private repo: AuthRepo) {}
  call: () => AsyncResult<ResponseMessage> = () => {
    return this.repo.logout();
  };
}

export default Logout;

File: blogapp_flutter/lib/features/authentication/domain/usecases/logout.dart
import 'package:blogapp_flutter/core/typedefs/async_result.dart';
import 'package:blogapp_flutter/core/typedefs/usecases.dart';

import '../../data/models/response_message.dart';
import '../repos/auth_repo.dart';

class Logout extends UsecaseWithoutParams<ResponseMessage> {
  final AuthRepo repo;

  Logout(this.repo);

  @override
  AsyncResult<ResponseMessage> call() async {
    return await repo.logout();
  }
}
